### DISCUSSION TITLE
[Feature] Natural language input for Date fields - Date of Birth, Editing the date of images etc.

### DISCUSSION THREAD

**Discussion by radh21301** [immich-app/immich#12690] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is the same feature request I made to immich-power-tools, and it has been implement in [v0.6.0](https://github.com/varun-raj/immich-power-tools/releases/tag/v0.6.0). It would be awesome if Immich can implement it natively as well!

So, right now we've got that calendar picker for DOB, which is fine. But what if we could also just type in dates however we want? Like "Jan 12 2004" or "15 Feb 2019" - you know, the way we actually say dates.

I was thinking it'd be super handy, especially for people who aren't used to specific date formats. Like, I always get mixed up with mm/dd/yyyy vs dd/mm/yyyy. This way, you could just type it how you remember it.

From my basic search with ChatGPT, looks like there are python libraries like 'dateutil' that can do this? The dev mentioned using a library called [chrono](https://github.com/wanasit/chrono) for power tools. Sounds like it wouldn't be too tricky to set up?

It's not a must-have, but it'd be a nice little extra that could make things smoother for everyone, especially folks who aren't super techy. What do you think?

**Additional info:**
The video that Varun, the dev of immich power tools shared. Note, as seen in the end of the video, there is an edge case where he types 1995 4th July, but after hitting enter it becomes July 4th, 2024. But it work really well for most use cases! pretty cool! :) The request is not necessarily to use the same library, but include something that can achieve this functionality.

https://github.com/user-attachments/assets/8b16c053-6259-44a3-9ede-12d52c548cc3

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by mertalev** [comment]
I'm a little cautious about this. I think time-related issues are the opposite of fun and an open-ended feature like this will inevitably invite more of them. The library also doesn't support a lot of the languages we do.

**Reply by radh21301** [reply]
Yeah. I see a lot of issues related to time/timezone coming up. But maybe we could provide a a reasonable/restrictive leeway with what can be typed in this field. I mean, looks like the crono library can accept things like Jan 13 1987  12:00:00 GMT-4:00 -> for example, we could possibly prevent these kind of inputs with Timezone and restrict to only Date and Time. We could also provide separate boxes to type Date and Time - so Immich has some control/restriction here as well?

I didn't think about the language aspect. Maybe there is a better library out there? Or we could possibly enable this capability only for certain languages (set in the Account Settings> App settings > Language? 

I dont want to assume, but I wonder what percentage of users that set English as the app language. I am not a native English speaker, but use English as the app language because it is much easier to type in english than in my native one. :) If only Immich had some analytics from its users :) 

p.s. I think there is always going to be some feature that is not supported in all languages immich supports correct? I mean CLIP does not work in all languages?

**Reply by mertalev** [reply]
As you say, it'd be better to reign in the supported input so we're not on the hook for everything the library supports. I imagine there might be some complexity in having a graceful fallback for unsupported languages, but it shouldn't be too bad.

The default CLIP model only supports English, but there are others that support 100+ languages.
